rust/src/lib/nm/query_apply/route.rs

```diff
@@ -37,14 +35,7 @@ fn is_nm_ip_route_removed(
     new_routes: &[NmIpRoute],
     cur_routes: &[NmIpRoute],
 ) -> bool {
-    for cur_route in cur_routes {
-        let mut cur_route_norm = cur_route.clone();
-        if cur_route_norm.table.is_none() {
-            cur_route_norm.table = Some(DEFAULT_TABLE_ID);
```

Comment:
I added this workaround to prevent the full activation on the existing connection, https://github.com/nmstate/nmstate/pull/2639, but I think that your approach is better, because you convert the predefined default values to None at creation time, which will guarantee that the route table, next-hop will have the consistent default value for new routes and current routes, therefore, the comparisons are always successful.